pkg/sql/schemachanger/scexec/scmutationexec/references.go

```diff
@@ -112,7 +112,11 @@ func (m *visitor) MakeAbsentCheckConstraintWriteOnly(
 		FromHashShardedColumn: op.FromHashShardedColumn,
 		ConstraintID:          op.ConstraintID,
 	}
-	return enqueueAddCheckConstraintMutation(tbl, ck)
+	if err = enqueueAddCheckConstraintMutation(tbl, ck); err != nil {
+		return err
+	}
+	tbl.Mutations[len(tbl.Mutations)-1].State = descpb.DescriptorMutation_WRITE_ONLY
```

Comment:
could you explain why we want to move the state to write_only here instead of doing it in an operation at some later point? maybe a comment would be helpful

Reply:
It actually does not matter even if we don't do it (and leave it in the DELETE_ONLY state); It's good that we do this (fast-forward it to WRITE_ONLY) bc that aligns with the concept "enforced" that we introduced, that is, such a constraint, at this time, is considered "enforced" and should better have its mutation state be WRITE_ONLY.

I've added a comment here.